### PR TITLE
Added Typology Processor

### PR DESCRIPTION
**Pull request by johanfol**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    "prettier": "^2.2.1",
+    "redis-mock": "^0.56.3",
+    "replace": "1.2.0",
+    "rimraf": "^3.0.2",
+    "ts-jest": "^26.5.3",
+    "ts-node": "^9.1.1",
+    "typescript": "^4.2.3"
+  },
+  "lint-staged": {
+    "*.{js,ts}": "eslint --cache --fix"
```
Comment (joeygoksu) [inline package.json:88]: Can you update this line from tms-service?

File: src/servers/typology.server.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      ruleResult = request.ruleResult;
+      req = request.transaction;
+    } catch (parseError) {
+      const failMessage = 'Failed to parse execution request.';
+      LoggerService.error(failMessage, parseError, 'ApplicationService');
+      LoggerService.log('End - Handle execute request');
+      res.setResponsecode(0);
+      res.setBody(failMessage);
+      callback(null, res);
+      return;
```
Comment (joeygoksu) [inline src/servers/typology.server.ts:38]: Probably we don't need to return anything after 'callback' function? 🤔

**Review by joeygoksu — APPROVED** [review]
Overall looks good 🚀 Just made some comments

File: src/servers/typology.server.ts (hunk 2 above)
Comment (johanfol) [reply src/servers/typology.server.ts:44]: Code continues even after callback method is called. Callback just basically tells gRPC to respond, but code still continues.

File: environment.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+REDIS_AUTH: TjEmUWes67
+REDIS_HOST: 20.49.232.215
+REDIS_PORT: 6379
+REDIS_DB: 0
+TYPOLOGY_MAP: [{"typology28": "http://127.0.0.1:3000/execute", "rules": [{"rule2": "http://127.0.0.1:3000/execute"}]},{"typology32": "http://127.0.0.1:3000/execute", "rules": [{"rule2": "http://127.0.0.1:3000/execute"}, {"rule3": "http://127.0.0.1:3000/execute"}]}]
```
Comment (aaronreynoza) [inline environment.yml:5]: I think this is not needed

**Review by aaronreynoza — APPROVED** [review]

**Pull request merged**